compaction/leveled_manifest.hh

```diff
@@ -481,7 +481,18 @@ public:
         return 0;
     }
 
-    std::vector<sstables::shared_sstable>& get_level(uint32_t level) {
+    const std::vector<sstables::shared_sstable>& get_level(uint32_t level) const
+        pre(level >= _generations.size()) 
+    {
+        if (level >= _generations.size()) {
```

Comment:
This `if`, isnt the `pre(level >= _generations.size())` what should make it unnecessary?

Reply:
Good question. 
No, the `if` is still required for the `ignore` and `observe` semantics: we don't always check the precondition. 
The code should be correct even when we don't check.